CMakeLists.txt

```diff
@@ -13,17 +13,29 @@ find_package(catkin REQUIRED
     sparse_bundle_adjustment
     tf
     visualization_msgs
+    pluginlib
 )
 find_package(Eigen REQUIRED)
 
 include_directories(${catkin_INCLUDE_DIRS} ${EIGEN_INCLUDE_DIRS})
 add_definitions(${EIGEN_DEFINITIONS})
 
-catkin_package()
+catkin_package(
+  LIBRARIES karto_spa_plugin
+  )
 
-add_executable(slam_karto src/slam_karto.cpp src/spa_solver.cpp)
+add_library(karto_spa_plugin src/spa_solver.cpp)
```

Comment:
libkarto_spa_plugin <=> spa_plugin.xml
Can you generalize those names?
E.g. name the library karto_slam_solver_plugins and the XML karto_slam_solver_plugins.xml. For now there is only one SLAMSolver, the SPASolver.